File: apps/mobile/lib/domain/usecases/school/delete_school_usecase.dart
import 'package:mobile/core/error/failures.dart';
import 'package:mobile/core/usecases/usecase.dart';
import 'package:mobile/core/utils/either.dart';
import 'package:mobile/domain/repositories/school/school_repository.dart';

class DeleteSchoolUseCase implements UseCase<Either<Failure, void>, NoParams> {
  final SchoolRepository repository;

  DeleteSchoolUseCase(this.repository);

  @override
  Future<Either<Failure, void>> call(NoParams params) async {
    return await repository.deleteSchool();
  }
}

File: services/user_service/src/domain/usecases/school/DeleteSchoolUseCase.ts
import { ISchoolRepository } from "../../repositories/ISchoolRepository";

export class DeleteSchoolUseCase {
  constructor(private schoolRepository: ISchoolRepository) {}

  async execute(id: string): Promise<boolean> {
    const school = await this.schoolRepository.findById(id);
    if (!school) {
      throw new Error("School not found");
    }

    return await this.schoolRepository.delete(id);
  }
}
